src/main/java/net/glowstone/block/data/IBlockData.java

```diff
@@ -0,0 +1,21 @@
+package net.glowstone.block.data;
+
+import net.glowstone.block.data.state.StateGenerator;
+import net.glowstone.block.data.state.StateValue;
+import org.bukkit.block.data.BlockData;
+
+import java.util.Set;
+
+public interface IBlockData extends BlockData {
+
+    Set<StateValue<?>> getStateValues();
```

Comment:
Should this be a map instead of a set? Seems like, quite often, we're fetching things by ID, which requires a scan. Admittedly, this won't be a huge performance impact, since there won't be many pieces of state, but it still would be good practice from a design perspective.

Reply:
Never knew about that design perspective. Just wanted it to be easy to maintain so I didnt allow for the possibility of the two IDs being unsynced by a plugin. 

Ill change it